packages/compass-preferences-model/src/provider.ts

```diff
@@ -1,18 +1,68 @@
 import { createContext, useContext } from 'react';
-import type { PreferencesAccess } from './';
-import { defaultPreferencesInstance } from '.';
+import type {
+  AllPreferences,
+  PreferenceStateInformation,
+  PreferencesAccess,
+  UserConfigurablePreferences,
+} from './';
+import { getDefaultsForAllPreferences } from './preferences-schema';
 export { usePreference, withPreferences } from './react';
 export { capMaxTimeMSAtPreferenceLimit } from './maxtimems';
 
+export class ReadOnlyPreferenceService implements PreferencesAccess {
+  private allPreferences: AllPreferences;
+  constructor(preferencesOverrides?: Partial<AllPreferences>) {
+    this.allPreferences = {
+      ...getDefaultsForAllPreferences(),
+      ...preferencesOverrides,
+    };
+  }
+
+  savePreferences() {
+    return Promise.resolve(this.allPreferences);
+  }
+
+  refreshPreferences() {
+    return Promise.resolve(this.allPreferences);
+  }
+
+  getPreferences() {
+    return this.allPreferences;
+  }
+
+  ensureDefaultConfigurableUserPreferences() {
+    return Promise.resolve();
+  }
+
+  getConfigurableUserPreferences() {
+    return Promise.resolve({} as UserConfigurablePreferences);
+  }
+
+  getPreferenceStates() {
+    return Promise.resolve({} as PreferenceStateInformation);
+  }
+
+  onPreferenceValueChanged() {
+    return () => {
+      // noop
+    };
+  }
+
+  createSandbox() {
+    return Promise.reject('Method not supported');
+  }
```

Comment:
I don't think we can guarantee that this will not be called somewhere in Compass easily as it's part of the interface and not optional, so I'd suggest doing the other option that Anna suggested, return a deep clone / new instance here instead of returning `this`, but not throw

Reply:
We can, but there are additional platform specific dependencies to remove from the class: implicit logging instance, everything related to global config and cli args handling. What we want is preferences class that truly doesn't have anything platform specifc in the implementation and then compass can implement a version that also handles global config files, cli arguments, ipc, persisting with user-data, none of those things will ever be part of the version used by data explorer

Reply:
To the original concern that Sergey pointed out - we might not actually need SandboxPreference anymore, after having this new simple preference service implementation. The only place that I noticed we use the sandbox preferences is tests (please correct if I missed something) but now our provider always gets an instance of our service implementation and our preference locator does not anymore return the `defaultPreferenceInstance`. So the need for creating sandbox is essentially gone now.

Did I miss something or is it making sense?

Reply:
Sandbox preferences are used in settings so that the changes can be made to the preferences taking all the logic of deriving the values into account and then applied to the parent preferences. They can't be "read only" because this is what is being changed by the user when clicking around in the settings modal

Reply:
Yea this makes sense now, sorry I did not look thoroughly how the sandbox was used in here. Ok so this completely disproves my idea of killing the sandbox as I was under wrong understanding that the sandbox interface was used just to aid in testing.

Reply:
> Honestly, if we want to use this in production, we should also consider to just keep using the regular Preferences class with a custom storage service.

@addaleax we chatted a bit with Himanshu and took a close look at the preferences class implementation, and I think we found a way to cover both your suggestion while making the interface more generalized and less platform specific without a lot of changes (there is not that much specific stuff there really), so Himanshu will be working on a follow up that incorporates both your feedback here and the requirement for a slightly less platform dependant implementation. Thank you for the feedback!